scripts/run_backend_tests.py

```diff
@@ -339,19 +341,28 @@ def main(args=None):
             raise Exception('Directory %s does not exist.' % directory)
         sys.path.insert(0, directory)
 
+    pythonpath_starting_index = 0
```

Comment:
This seems really complex. Why not just do `sys.path.insert(0, ...)`? (You'll see this elsewhere in our codebase.)

Reply:
Even if this code has some reason, it should have some comment explaining what it does, because I'm not able to understand that.

Reply:
The code became complex because we needed to get `PYTHONPATH` set so that child processes (which are create for each test) has access to the coverage module. I've found another way that leverages our existing path insertion methods.